ledger-service/http-json/src/main/scala/com/digitalasset/http/LedgerClientJwt.scala

```diff
@@ -80,7 +80,21 @@ object LedgerClientJwt {
     LedgerOffset(LedgerOffset.Value.Boundary(LedgerOffset.LedgerBoundary.LEDGER_END))
 
   def getCreatesAndArchivesSince(client: LedgerClient): GetCreatesAndArchivesSince =
-    (jwt, filter, offset, terminates) =>
-      client.transactionClient
-        .getTransactions(offset, terminates.toOffset, filter, verbose = true, token = bearer(jwt))
+    (jwt, filter, offset, terminates) => {
+      val end = terminates.toOffset
+      if (skipRequest(offset, end))
+        Source.empty[Transaction]
+      else
+        client.transactionClient
+          .getTransactions(offset, terminates.toOffset, filter, verbose = true, token = bearer(jwt))
+    }
+
+  private def skipRequest(start: LedgerOffset, end: Option[LedgerOffset]): Boolean =
+    (start.value, end.map(_.value)) match {
+      case (LedgerOffset.Value.Absolute(s), Some(LedgerOffset.Value.Absolute(e))) =>
+        // the same comparision as in here
+        // https://github.com/digital-asset/daml/blob/e2a5b264750dd9d96a50c8fd180a08d6f2eb0860/ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala#L275
+        s.toLong >= e.toLong
```

Comment:
This comparison is appropriate for the server but not for the client; see

https://github.com/digital-asset/daml/blob/cb465837efd2ff351a771c90b5b6bfa22956d1e2/ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/ledger_offset.proto#L17-L22

If one offset is in one format and the other in another format, make an arbitrary (but consistent) choice as to whether integer offsets are less than or greater than string-num-num triple offsets.

Reply:
If you like, we can then Gen offsets of both varieties and test the Order laws, since no sandbox integration test will properly cover the triple case.

Reply:
`Absolute` offset has `number`

```
  final case class Absolute(val value : scala.Predef.String) extends scala.AnyRef with com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset.Value with scala.Product with scala.Serializable {
...
      override def absolute : scala.Option[scala.Predef.String] = { /* compiled code */ }
      override def number : scala.Int = { /* compiled code */ }
    }
```
Is that guaranteed to provide proper conversion? @gerolf-da

Reply:
// It is either a string representing an ever-increasing integer, or
    // a composite string containing ``<block-hash>-<block-height>-<event-id>``; ordering
    // requires comparing numerical values of the second, then the third element.

do we have a utility that can compare two offset?

Reply:
> Is that guaranteed to provide proper conversion?

No, `number` is the index in the `oneof`.

Reply:
> If you like, we can then Gen offsets of both varieties and test the Order laws, since no sandbox integration test will properly cover the triple case.

yep, that is what I was planning todo, all server-side code tests with `offset.value.toInt` `Ordering`. I am creating the implementation that would handle both `Int` and `<block-hash>-<block-height>-<event-id>`.

Reply:
I don't think we need to handle comparison between Int offset and composite offset. It is supposed to be consistent for the ledger implementation.